### PR TITLE
Create a WidgetWrapper component for fetching data

### PR DESCRIPTION
**Pull request by siemiatj**
Second attempt after doing a thorough manual testing. Related to #9914

### CODE REVIEW

**Review by petrican — CHANGES_REQUESTED** [review]

**Comment by petrican** [comment]
Looks fine, left some comments. I would request a 2nd functional smoke tests before merging this into master from another developer willing to test this of from Daniela but doubt she is available. Maybe ask Phillip who might be available to test.

File: frontend/src/components/widget/RawWidget.js [hunk 1]
```diff
@@ -449,19 +454,19 @@ export class RawWidget extends Component {
       updateHeight,
       rowIndex,
     } = this.props;
-    let widgetValue = data != null ? data : widgetData[0].value;
+    let tabIndex = this.props.tabIndex;
     const { isEdited, charsTyped } = this.state;
 
+    let widgetValue = data != null ? data : widgetData[0].value;
```
[the remark this answers is not in the document]
Comment (siemiatj) [reply frontend/src/components/widget/RawWidget.js:460]: Nope. This tests for values not set - `null` and `undefined`. `''` and `0` will still work.

File: frontend/src/containers/WidgetWrapper.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+      break;
+    case 'modal':
+    case 'element':
+      widgetData = getWidgetData(state, isModal, layoutId);
+      fieldsCopy = getWidgetFields(state, isModal, layoutId);
+
+      break;
+    case 'process':
+      widgetData = getProcessWidgetData(state, true, layoutId);
```
[the remark this answers is not in the document]
Comment (siemiatj) [reply frontend/src/containers/WidgetWrapper.js:78]: It's process's widget data, not widget's process data.

File: frontend/src/components/window/Element.js [hunk 3]
```diff
@@ -1,42 +1,57 @@
...
+      columnIndex,
+    } = this.props;
 
     const autoFocus = isFocused && elementIndex === 0;
-    const widgetData = elementLayout.fields.map(
-      (item) => data[item.field] || -1
-    );
     const fieldName = elementLayout.fields ? elementLayout.fields[0].field : '';
-    const relativeDocId = data.ID && data.ID.value;
+    const layoutId = `${sectionIndex}_${columnIndex}_${elementGroupIndex}_${elementsLineIndex}_${elementIndex}`;
```
[the remark this answers is not in the document]
Comment (siemiatj) [reply frontend/src/components/window/Element.js:31]: Yeah but in this case this is the only place we're using it so extracting this to a separate function wouldn't benefit us in any way.

**Comment by petrican** [comment]
lgtm. Thanks!

**Review by petrican — APPROVED** [review]

**Pull request merged**